lldb/source/Plugins/OperatingSystem/SwiftTasks/OperatingSystemSwiftTasks.cpp

```diff
@@ -0,0 +1,155 @@
+
+LLDB_PLUGIN_DEFINE(OperatingSystemSwiftTasks)
+
+void OperatingSystemSwiftTasks::Initialize() {
+  PluginManager::RegisterPlugin(GetPluginNameStatic(),
+                                GetPluginDescriptionStatic(), CreateInstance,
+                                nullptr);
+}
+
+void OperatingSystemSwiftTasks::Terminate() {
+  PluginManager::UnregisterPlugin(CreateInstance);
+}
+
+OperatingSystem *OperatingSystemSwiftTasks::CreateInstance(Process *process,
+                                                           bool force) {
+  if (!process || !process->GetTarget().GetSwiftUseTasksPlugin())
+    return nullptr;
+
+  Log *log = GetLog(LLDBLog::OS);
+  std::optional<uint32_t> concurrency_version =
+      SwiftLanguageRuntime::findConcurrencyDebugVersion(*process);
+  if (!concurrency_version) {
+    LLDB_LOG(log,
+             "OperatingSystemSwiftTasks: did not find concurrency module.");
+    return nullptr;
+  }
+
+  if (*concurrency_version <= 1)
+    return new OperatingSystemSwiftTasks(*process);
+  LLDB_LOGF(log,
+            "OperatingSystemSwiftTasks: got a concurrency version symbol of %u",
+            *concurrency_version);
+  return nullptr;
+}
+
+llvm::StringRef OperatingSystemSwiftTasks::GetPluginDescriptionStatic() {
+  return "Operating system plug-in converting Swift Tasks into Threads.";
+}
+
+OperatingSystemSwiftTasks::~OperatingSystemSwiftTasks() = default;
+
+OperatingSystemSwiftTasks::OperatingSystemSwiftTasks(
+    lldb_private::Process &process)
+    : OperatingSystem(&process) {
+  size_t ptr_size = process.GetAddressByteSize();
+  // Offset of a Task ID inside a Task data structure, guaranteed by the ABI.
+  // See Job in swift/RemoteInspection/RuntimeInternals.h.
+  m_task_id_offset = 4 * ptr_size + 4;
+}
+
+bool OperatingSystemSwiftTasks::UpdateThreadList(ThreadList &old_thread_list,
+                                                 ThreadList &core_thread_list,
+                                                 ThreadList &new_thread_list) {
+  Log *log = GetLog(LLDBLog::OS);
+  LLDB_LOG(log, "OperatingSystemSwiftTasks: Updating thread list");
+
+  for (const ThreadSP &real_thread : core_thread_list.Threads()) {
+    std::optional<uint64_t> task_id = FindTaskId(*real_thread);
+
+    // If this is not a thread running a Task, add it to the list as is.
+    if (!task_id) {
+      new_thread_list.AddThread(real_thread);
+      LLDB_LOGF(log,
+                "OperatingSystemSwiftTasks: thread %" PRIx64
+                " is not executing a Task",
+                real_thread->GetID());
+      continue;
+    }
+
+    // Mask higher bits to avoid conflicts with core thread IDs.
+    uint64_t masked_task_id = 0x0000000f00000000 | *task_id;
+
+    ThreadSP swift_thread = [&]() -> ThreadSP {
+      // If we already had a thread for this Task in the last stop, re-use it.
+      if (ThreadSP old_thread = old_thread_list.FindThreadByID(masked_task_id);
+          IsOperatingSystemPluginThread(old_thread))
```

Comment:
First time I'm seeing this syntax :p Don't you want to run the second part of the if condition only if the first one if valid ?
```suggestion
      if ((ThreadSP old_thread = old_thread_list.FindThreadByID(masked_task_id)) && IsOperatingSystemPluginThread(old_thread))
```

Reply:
`IsOperatingSystemPluginThread` is designed to take a pointer argument, so it knows how to handle nullptrs.

(btw the suggested code doesn't compile :P )